x/username/username.go

```diff
@@ -0,0 +1,73 @@
+package username
+
+import (
+	"encoding/json"
+	"regexp"
+
+	"github.com/iov-one/weave/errors"
+)
+
+// Username represents a name registered on certain domain. A valid username is
+// in format <name>*<domain>. Username is case sensitive.
+type Username string
+
+// ParseUsername returns a valid username instance, extracted from given string
+// representation.
+func ParseUsername(s string) (Username, error) {
```

Comment:
I would store these two, name and domain, as two items in the db. And only combine in the ui or cli. Right now we have no domain support, but I could see an easy migration path with this implementation. Much easier to update and adapt in the future

Reply:
Relevant discussion https://github.com/iov-one/weave/issues/767#issuecomment-501987100

Do you have any use case in mind that storing those values separately makes easier? Right now, the `<name*domain>` string is used only as the token key. It is not stored as a value inside of the token. Do you want to store then additionally (in other words, add it to the model)?

Reply:
Hmmm... fair enough. Leave it as is